### PR TITLE
Update changelog for a 23.1.0 release

### PR DESCRIPTION
**Pull request by ismail-syed**
🚢🚢🚢

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -361,7 +363,8 @@ Example:
 
 Changes were originally tracked in Shopify's [JavaScript monorepo](https://github.com/Shopify/javascript/blob/f10bf7ddbdae07370cfe7c94617c450257731552/CHANGELOG.md).
 
-[Unreleased]: https://github.com/Shopify/eslint-plugin-shopify/compare/v23.0.0...HEAD
+[Unreleased]: https://github.com/Shopify/eslint-plugin-shopify/compare/v23.1.0...HEAD
+[Unreleased]: https://github.com/Shopify/eslint-plugin-shopify/compare/v23.0.0...v23.1.0
```
Comment (GoodForOneFare) [inline CHANGELOG.md]: `23.1.0`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 
-## Unreleased
+<!-- ## Unreleased -->
+
+## [23.0.0] - 2018-08-02
 
 ### Fixed
 * Updated `typescript-eslint-parser` dependency to version 17.0.1 in order to support TypeScript 3.
```
Comment (GoodForOneFare) [inline CHANGELOG.md]: Why did we stop linking to PRs?

Why isn't there an entry for, `Removed default prettier configurations`?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 
-## Unreleased
+<!-- ## Unreleased -->
+
+## [23.0.0] - 2018-08-02
```
Comment (GoodForOneFare) [inline CHANGELOG.md]: `23.1.0`

File: CHANGELOG.md (hunk 2 above)
Comment (GoodForOneFare) [reply CHANGELOG.md]: Oh, there's an entry for the prettier configs.  Github is just choosing not to show it :/

File: CHANGELOG.md (hunk 1 above)
Comment (ismail-syed) [reply CHANGELOG.md]: Had these changes in my editor.
_Forgot to hit save..._

File: CHANGELOG.md (hunk 2 above)
Comment (ismail-syed) [reply CHANGELOG.md]: Looks like we forgot to link PRs in those changelog entries. 

That entry is covered by the next line: 
> `plugin:shopify/prettier` and `plugin:shopify/typescript-prettier` now defer Prettier's config to project's `.prettierrc`.

But that additional context you suggested is definitely helpful to readers.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
 
-## Unreleased
+<!-- ## Unreleased -->
+
+## [23.1.0] - 2018-08-02
 
 ### Fixed
-* Updated `typescript-eslint-parser` dependency to version 17.0.1 in order to support TypeScript 3.
-* `plugin:shopify/prettier` and `plugin:shopify/typescript-prettier` now defer Prettier's config to project's `.prettierrc`.
+* Updated `typescript-eslint-parser` dependency to version 17.0.1 in order to support TypeScript 3. [#121](https://github.com/Shopify/eslint-plugin-shopify/pull/121)
```
Comment (TzviPM) [inline CHANGELOG.md]: pr references should be enclosed in parentheses for consistency, no?

**Review by TzviPM — APPROVED** [review]
:shipit:

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
-## Unreleased
+<!-- ## Unreleased -->
+
+## [23.1.0] - 2018-08-02
 
 ### Fixed
-* Updated `typescript-eslint-parser` dependency to version 17.0.1 in order to support TypeScript 3.
-* `plugin:shopify/prettier` and `plugin:shopify/typescript-prettier` now defer Prettier's config to project's `.prettierrc`.
+* Updated `typescript-eslint-parser` dependency to version 17.0.1 in order to support TypeScript 3. ([#121](https://github.com/Shopify/eslint-plugin-shopify/pull/121))
+* Removed default prettier configurations. `plugin:shopify/prettier` and `plugin:shopify/typescript-prettier` now defer Prettier's config to project's `.prettierrc`. ([#121](https://github.com/Shopify/eslint-plugin-shopify/pull/121))
```
Comment (GoodForOneFare) [inline CHANGELOG.md]: The apostrophe is indicating a singular project, but the sentence reads like it's talking about many projects.

Use either `a project's .prettierrc` or switch up to a address all projects with `projects' .prettierrcs`.

**Review by cartogram — APPROVED** [review]

**Review by GoodForOneFare — APPROVED** [review]

**Pull request merged**